data/src/config/server.rs

```diff
@@ -99,10 +101,44 @@ impl Server {
             connection::Security::Unsecured
         };
 
+        let proxy = self.proxy.clone().map(|p| p.into());
+
         connection::Config {
             server: &self.server,
             port: self.port,
             security,
+            proxy,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum ProxyConfigType {
+    Socks5,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct ProxyConfig {
+    #[serde(rename = "type")]
+    pub proxy_type: ProxyConfigType,
+    pub host: String,
+    pub port: u16,
+    #[serde(default)]
+    pub username: String,
+    #[serde(default)]
+    pub password: String,
+}
```

Comment:
Let's create a new config::proxy module and put these there. You can then just call it Proxy and Kind (proxy::Kind)